File: API/Entity/Announcement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entity
{
    public class Announcement
    {
        public long Id { get; set; }
        [Required]
        [MinLength(6)]
        [MaxLength(100)]
        public string Name { get; set; }
        [Required]
        [MinLength(10)]
        [MaxLength(1000)]
        public string Description { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public float Duration { get; set; }
        [Required]
        public bool IsActive { get; set; } = true;
        [Required]
        public bool IsAccepted { get; set; } = false;
        [Required]
        public DateTime DateCreated { get; set; } = DateTime.Now;
        [Required]
        public long AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        [Required]
        public long AddressId { get; set; }
        public Address Address { get; set; }
        public ICollection<Photo> Photos { get; set; }
        public ICollection<Comment> Comments { get; set; }

    }
}

File: client/src/app/models/announcement.ts
import { Address } from "./Address";
import { Photo } from "./photo";

export interface Announcement {
    id: number;
    name: string;
    description: string;
    startDate: Date;
    duration: number;
    photoUrl: string;
    dateCreated: Date;
    user: AnnouncementCreator;
    address: Address;
    photos: Photo[];
}

export interface AnnouncementCreator {
  id: number;
  userName: string;
  firstName: string;
  lastName: string;
  gender: string;
  created: Date;
  lastActive: Date;
}
